### PR TITLE
[ISSUE-750][Refactor] Add UserIdentifier as a field of file info 

### PR DESCRIPTION
**Pull request by nafiyAix**
### What changes were proposed in this pull request?
Add user identifier as a field of file info, and change related openSteam() method and OpenStream class code to fit with UserIdentifier.

### Why are the changes needed?
for better collect file's info by userIdentifier.

### Related issues.
#750 

### How was this patch tested?

/cc @AngersZhuuuu 

/assign @main-reviewer

### CODE REVIEW

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java [hunk 1]
```diff
@@ -162,7 +163,12 @@ public int getSortedSize() {
   }
 
   public FileInfo openStream(
-      String shuffleKey, String fileName, FileInfo fileInfo, int startMapIndex, int endMapIndex)
+      String shuffleKey,
+      String fileName,
+      FileInfo fileInfo,
+      UserIdentifier userIdentifier,
+      int startMapIndex,
+      int endMapIndex)
```
Comment (AngersZhuuuu) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java:171]: We have passed a `fileInfo` here

**Comment by nafiyAix** [comment]
FYI: Besides revert previous openstream related change, also modify the logic of `userResourceConsumptionSnapshot()` after removing `userShuffleKeys` from StorageManager, and add a return to break loop when recovering data from levelDB and there is no more levelDB key could be matched .

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Comment by waitinfuture** [comment]
I think we can cache UserIdentifiers in the process of reloading from leveldb to  avoid too many duplicate objects of the same user.

**Comment by nafiyAix** [comment]
> I think we can cache UserIdentifiers in the process of reloading from leveldb to avoid too many duplicate objects of the same user.

Agree, will implement it in #767

**Pull request merged**